### PR TITLE
Create SonarQube Analyzer Test Commons using "sonar-rule-unit-test" from TSQL

### PR DESCRIPTION
**Pull request by alban-auzeill**
- Move root file to "commons" folder
- Import "sonar-rule-unit-test" into "test-commons" folder
- Add documentation in "test-commons/README.md"

### CODE REVIEW

File: test-commons/src/main/java/com/sonarsource/checks/verifier/PreciseLocationParser.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+  // line adjustment, ex: @+1
+  public static final String LINE_ADJUSTMENT = "(?:@(?<lineAdjustment>(?<relativeAdjustment>[+-])?\\d++))?";
+
+  private static final Pattern LOCATION_REGEX = Pattern.compile(" *+" +
+  // highlighted range, ex: ^^^^ |OR| ^^^@12 |OR| ^^^@-2
+    "(?<range>\\^(?:\\[(?<params>[^\\]]++)\\]|\\^++)?)" + LINE_ADJUSTMENT +
+    // count, ex: 3 |OR| direction, ex: < |OR| direction with index, ex: < 1 |OR| direction and flowId, ex: < 2.1
+    " *+(?:(?<count>\\d++)|(?:(?<direction><|>) *+((?<majorIndex>\\d++)(\\.(?<minorIndex>\\d++))?)?))?" +
+    // message, ex: {{msg}}
```
Comment (SonarTech) [inline test-commons/src/main/java/com/sonarsource/checks/verifier/PreciseLocationParser.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: test-commons/src/main/java/com/sonarsource/checks/verifier/NoncompliantCommentParser.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+public final class NoncompliantCommentParser {
+
+  private static final Pattern START_WITH_NONCOMPLIANT = Pattern.compile("^(?i) *+Noncompliant");
+
+  private static final Pattern NON_COMPLIANT_REGEX = Pattern.compile("(?i) *+Noncompliant" + LINE_ADJUSTMENT
+  // issue count, ex: 2
+    + "(?: ++(?<issueCount>\\d++))?"
+    + " *+"
+    // messages, ex: {{msg1}} {{msg2}}
```
Comment (SonarTech) [inline test-commons/src/main/java/com/sonarsource/checks/verifier/NoncompliantCommentParser.java]: MAJOR This block of commented-out lines of code should be removed. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [ProfileDefinitionReader.java#L39](https://github.com/SonarSource/sonar-analyzer-commons/blob/3f62c534a74ded3728fa824647ec98cc80a4d145/commons/src/main/java/org/sonarsource/analyzer/commons/ProfileDefinitionReader.java#L39): Remove this use of "RuleFinder"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [ProfileDefinitionReader.java#L41](https://github.com/SonarSource/sonar-analyzer-commons/blob/3f62c534a74ded3728fa824647ec98cc80a4d145/commons/src/main/java/org/sonarsource/analyzer/commons/ProfileDefinitionReader.java#L41): Remove this use of "RuleFinder"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [ProfileDefinitionReader.java#L39](https://github.com/SonarSource/sonar-analyzer-commons/blob/b8498c8fd77357db1e091506e139c6261090f950/commons/src/main/java/org/sonarsource/analyzer/commons/ProfileDefinitionReader.java#L39): Remove this use of "RuleFinder"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [ProfileDefinitionReader.java#L41](https://github.com/SonarSource/sonar-analyzer-commons/blob/b8498c8fd77357db1e091506e139c6261090f950/commons/src/main/java/org/sonarsource/analyzer/commons/ProfileDefinitionReader.java#L41): Remove this use of "RuleFinder"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by SonarTech** [comment]
SonarQube analysis reported 4 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 2 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [ProfileDefinitionReader.java#L39](https://github.com/SonarSource/sonar-analyzer-commons/blob/ade6a1e64825873b8632ae60a99a0176a418cbe2/commons/src/main/java/org/sonarsource/analyzer/commons/ProfileDefinitionReader.java#L39): Remove this use of "RuleFinder"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)
1. ![MINOR][MINOR] [ProfileDefinitionReader.java#L41](https://github.com/SonarSource/sonar-analyzer-commons/blob/ade6a1e64825873b8632ae60a99a0176a418cbe2/commons/src/main/java/org/sonarsource/analyzer/commons/ProfileDefinitionReader.java#L41): Remove this use of "RuleFinder"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: test-commons/src/main/java/com/sonarsource/checks/verifier/LineIssues.java [hunk 3]
```diff
@@ -167,14 +167,24 @@ private void appendLocations(StringBuilder out) {
       Map<Integer, PerLineLocationWriter> writerPerLine = new TreeMap<>();
       for (PreciseLocation location : locations) {
         writerPerLine.computeIfAbsent(
-          location.range.line, key -> new PerLineLocationWriter(String.format("%03d: ", key), testFile.lineWithoutComment(key))).add(location);
+          location.range.line, key -> new PerLineLocationWriter(String.format("%03d: ", key), reportLineAt(key))).add(location);
```
Comment (inverno) [inline test-commons/src/main/java/com/sonarsource/checks/verifier/LineIssues.java]: @alban-auzeill can we use something more explicit than `line == 0` to spot file-issues?

File: test-commons/src/main/java/com/sonarsource/checks/coverage/UtilityClass.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
+/*
```
Comment (inverno) [inline test-commons/src/main/java/com/sonarsource/checks/coverage/UtilityClass.java]: @alban-auzeill why is UtilityClass in `main` instead of `test`?

**Pull request merged**